### PR TITLE
NewAnimalWizard updates for non-primate bulk entry

### PR DESCRIPTION
**Pull request by spamhurts**
#### Rationale
NewAnimalWizard should allow bulk ID creation for non-primate species

#### Related Pull Requests
none

#### Changes
1. Refactored protocolLookup query to exclude protocols that would exceed the maximum numbers
   allowed to be assigned. 
2. Refactored bulk entry for hamsters to bulk entry for non-primate species
3. Added isNonPrimate to animal state data
4. Changed CurrentSpeciesLookup to include existing species check from 10 years ago instead of 5

### CODE REVIEW

File: snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx [hunk 1]
```diff
@@ -105,11 +107,11 @@ loadLists() {
   }
 loadListsForSpecies = selectedSpecies => {
     const lists = {}
-
+    const newAssignments = this.state.numAnimals ? this.state.numAnimals : 1
```
Comment (labkey-martyp) [inline snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx]: `const newAssignments = this.state.numAnimals ?? 1`
or
`const newAssignments = this.state.numAnimals || 1`

is a little more succinct. Whether 0 is valid or not is the difference.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

File: snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx [hunk 2]
```diff
@@ -164,25 +192,36 @@ handleSpeciesChange = selectedSpecies => {
...
           species: selectedSpecies,
+          isNonPrimate: nonPrimate,
         },
       }))
       this.loadListsForSpecies(selectedSpecies)
     }
   };
 handleNumAnimalChange = value => {
-    this.setState(
+  const newAssignments = this.state.numAnimals ? this.state.numAnimals : 1
```
Comment (labkey-martyp) [inline snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx]: `const newAssignments = this.state.numAnimals ?? 1`
or
`const newAssignments = this.state.numAnimals || 1`

**Review by labkey-martyp — APPROVED** [review]
Looks good

File: snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx (hunk 1 above)
Comment (spamhurts) [reply snprc_ehr/src/client/NewAnimalPage/NewAnimalPage.jsx]: Good catch - thanks!

**Pull request merged**